tests/rgen/Microsoft.Macios.Generator.Tests/Emitters/BindingSyntaxFactoryTests.cs

```diff
@@ -0,0 +1,861 @@
+
+		[Field<Property> (""kCGColorSpaceGenericGray"")]
+		public partial CGColorSpaceGenericGray GenericGray { get; }
+
+	}
+}
+";
+
+			yield return [
+				ushortEnumFieldProperty,
+				"Dlfcn.GetUInt16 (Libraries.CoreGraphics.Handle, \"kCGColorSpaceGenericGray\");"
+			];
+
+			const string intEnumFieldProperty = @"
+using System;
+using Foundation;
+using CoreMedia;
+using ObjCBindings;
+
+namespace CoreGraphics {
+
+	public enum CGColorSpaceGenericGray : int {
+		First, 
+		Second,
+	}
+
+	[BindingType<Class>]
+	public partial class CGColorSpaceNames {
+
+		[Field<Property> (""kCGColorSpaceGenericGray"")]
+		public partial CGColorSpaceGenericGray GenericGray { get; }
+
+	}
+}
+";
+
+			yield return [
+				intEnumFieldProperty,
+				"Dlfcn.GetInt32 (Libraries.CoreGraphics.Handle, \"kCGColorSpaceGenericGray\");"
+			];
+
+			const string uintEnumFieldProperty = @"
+using System;
+using Foundation;
+using CoreMedia;
+using ObjCBindings;
+
+namespace CoreGraphics {
+
+	public enum CGColorSpaceGenericGray : uint {
+		First, 
+		Second,
+	}
+
+	[BindingType<Class>]
+	public partial class CGColorSpaceNames {
+
+		[Field<Property> (""kCGColorSpaceGenericGray"")]
+		public partial CGColorSpaceGenericGray GenericGray { get; }
+
+	}
+}
+";
+
+			yield return [
+				uintEnumFieldProperty,
+				"Dlfcn.GetUInt32 (Libraries.CoreGraphics.Handle, \"kCGColorSpaceGenericGray\");"
+			];
+
+			const string longEnumFieldProperty = @"
+using System;
+using Foundation;
+using CoreMedia;
+using ObjCBindings;
+
+namespace CoreGraphics {
+
+	public enum CGColorSpaceGenericGray : long {
+		First, 
+		Second,
+	}
+
+	[BindingType<Class>]
+	public partial class CGColorSpaceNames {
+
+		[Field<Property> (""kCGColorSpaceGenericGray"")]
+		public partial CGColorSpaceGenericGray GenericGray { get; }
+
+	}
+}
+";
+
+			yield return [
+				longEnumFieldProperty,
+				"Dlfcn.GetInt64 (Libraries.CoreGraphics.Handle, \"kCGColorSpaceGenericGray\");"
+			];
+
+			const string ulongEnumFieldProperty = @"
+using System;
+using Foundation;
+using CoreMedia;
+using ObjCBindings;
+
+namespace CoreGraphics {
+
+	public enum CGColorSpaceGenericGray : ulong {
+		First, 
+		Second,
+	}
+
+	[BindingType<Class>]
+	public partial class CGColorSpaceNames {
+
+		[Field<Property> (""kCGColorSpaceGenericGray"")]
+		public partial CGColorSpaceGenericGray GenericGray { get; }
+
+	}
+}
+";
+
+			yield return [
+				ulongEnumFieldProperty,
+				"Dlfcn.GetUInt64 (Libraries.CoreGraphics.Handle, \"kCGColorSpaceGenericGray\");"
+			];
+
+			const string cmTimeFieldProperty = @"
+using System;
+using Foundation;
+using CoreMedia;
+using ObjCBindings;
+
+namespace CoreGraphics {
+
+	[BindingType<Class>]
+	public partial class CGColorSpaceNames {
+
+		[Field<Property> (""kCGColorSpaceGenericGray"")]
+		public partial CMTime GenericGray { get; }
+
+	}
+}
+";
+
+			yield return [
+				cmTimeFieldProperty,
+				"*((CoreMedia.CMTime*) Dlfcn.dlsym (Libraries.CoreGraphics.Handle, \"kCGColorSpaceGenericGray\"));"
```

Comment:
I think this could be:

```suggestion
				"Dlfcn.GetStruct<CMTime> (Libraries.CoreGraphics.Handle, \"kCGColorSpaceGenericGray\"));"
```

Reply:
Looking at bgen we have:  
```csharp
./AVFoundation/AVCaptureDevice.g.cs:2684:                               return *((global::CoreMedia.CMTime *) Dlfcn.dlsym (Libraries.AVFoundation.Handle, "AVCaptureExposureDurationCurrent"));
./AVFoundation/AVCaptureDevice.g.cs:2825:                               return *((AVCaptureWhiteBalanceGains *) Dlfcn.dlsym (Libraries.AVFoundation.Handle, "AVCaptureWhiteBalanceGainsCurrent"));
```

Reply:
we can always get things working with the current machinery and move to better API once we can validate the changes

Reply:
@rolfbjarne @dalexsoto FYI https://github.com/xamarin/xamarin-macios/issues/21982